### PR TITLE
v4 NPE guard in MapType validate

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
A rather edge case for mgmt compute.

`ManagedServiceIdentity` actually need add a `key -> null` in UPDATE to remove a user identity. Hence value in Map could be null.

### CODE REVIEW

File: vanilla-tests/src/main/java/fixtures/modelflattening/AutoRestResourceFlatteningTestService.java [hunk 1]
```diff
@@ -366,7 +366,7 @@ public void putWrappedArray(List<WrappedProduct> resourceArray) {
             return Mono.error(new IllegalArgumentException("Parameter this.getHost() is required and cannot be null."));
         }
         if (resourceDictionary != null) {
-            resourceDictionary.values().forEach(e -> e.validate());
+            resourceDictionary.values().forEach(e -> {if (e != null) {e.validate();}});
```
Comment (jianghaolu) [inline vanilla-tests/src/main/java/fixtures/modelflattening/AutoRestResourceFlatteningTestService.java:369]: nit: space after { and before }

Comment (weidongxu-microsoft) [reply vanilla-tests/src/main/java/fixtures/modelflattening/AutoRestResourceFlatteningTestService.java:369]: Fixed. checkstyle actually allows code block in single line.
